### PR TITLE
GH-1617: Create warnings for variables with values of optional chained expressions

### PR DESCRIPTION
**Pull request by mmews-n4**
see #1617

### CODE REVIEW

File: tests/org.eclipse.n4js.xpect.tests/model/validation/dataflow/NullUndefined_May_OptionalChaining.n4js.xt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+/*	XPECT_SETUP org.eclipse.n4js.xpect.tests.N4JSXpectTest 
+	IssueConfiguration {
+		IssueCode "DFG_NULL_DEREFERENCE" {enabled=true}
+	}
+	END_SETUP
+ */
+
+class C {
+	field: C;
+}
```
Comment (mor-n4) [inline tests/org.eclipse.n4js.xpect.tests/model/validation/dataflow/NullUndefined_May_OptionalChaining.n4js.xt]: You might wanna add a test case for the "null not passed through optional chaining" rule:
```javascript
function f() {
	let c : C = new C();
	if (true) {
		c = null;
	}
	// XPECT warnings --> "Variable c may be null" at "x"
	c.field;

	let x = c?.field;
	// XPECT warnings --> "Variable x may be undefined" at "x"
	x.toString();
}
```

File: tests/org.eclipse.n4js.xpect.tests/model/validation/dataflow/NullUndefined_May_OptionalChaining.n4js.xt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	// XPECT warnings --> "Variable x may be undefined" at "x"
+	x.toString();
+}
+
+
+function f5() {
+	let c : C;
+	let x = c?.field;
+	// XPECT warnings --> "Variable x may be undefined" at "x"
+	x.toString();
```
Comment (mor-n4) [inline tests/org.eclipse.n4js.xpect.tests/model/validation/dataflow/NullUndefined_May_OptionalChaining.n4js.xt]: I think in this case it would make sense to warn with `Variable x is(!) undefined`.

**Comment by mmews-n4** [comment]
Good find. I improved the analysis to also follow the data flow of the target of optional chaining.
